### ISSUE TITLE
Regression from aptly_1.0.1 -> aptly_1.1.0 with aptly mirror update

### ISSUE THREAD

**Issue by pmorch** [smira/aptly#641]
## Detailed Description

With aptly 1.0.1 if I do

    aptly mirror update foobar

and it fails before finishing, then if I retry, it (re)uses the packages that it has already downloaded and only tries to download remaining packages.

With 1.1.0 (and 1.1.1), if an `aptly mirror update` fails, then retrying downloads everything again from scratch.

## Context

That is really annoying (also for mirrors?), because mirroring debian jessie for just i386 and amd64 is 120GB+ and if that fails (seen in real life but also tested with CTRL-C) after having downloaded 100GB, then it needs to download 120GB all over over again, instead of just the remaining 20GB. After 4 failed attempts at downloading 120GB, I have seen a ~/.aptly taking up 400GB disk space. (aptly db cleanup does fix this though).

I've tried to read [releases info](https://github.com/smira/aptly/releases) but there was no mention of this behaviour. So I'd like to ask: Is this a feature or a bug?

If its a feature, I'd like to understand why, because it seems like a waste of bandwidth, time and disk space. I didn't get any answer for 19 days to [my post in the google group](https://groups.google.com/d/topic/aptly-discuss/Eph83LF0T8w/discussion), so I'm writing up this issue here.

## Your Environment

Occurs in 1.1.0 and 1.1.1 on debian jessie

**Comment by smira** [comment]
Copying from mailing list:

```

Yes, if you cancel it on your own (but not if download fails for some files), aptly won't be able to keep already downloaded files.

This is great question, it was compromise between allowing other commands while download is going and allowing retries.

One good thing would be to save progress on ^C. Certainly something for me to consider for the next release.

Thanks for reporting that!
```

**Comment by smira** [comment]
Should be fixed with #545

**Comment by smira** [comment]
Should be available now in nightly build `aptly_1.1.1+75+g9cb2a30`
